packages/xo-server-auth-ldap/src/index.js

```diff
@@ -244,6 +319,22 @@ class AuthLdap {
             `successfully bound as ${entry.dn} => ${username} authenticated`
           )
           logger(JSON.stringify(entry, null, 2))
+
+          const idAttribute = this._usersConfig?.idAttribute
+          if (idAttribute === undefined) {
+            // Support legacy config
+            await this._xo.registerUser(undefined, username)
+          } else {
+            const ldapId = entry[idAttribute]
+            const user = await this._xo.registerUser2('ldap', {
+              user: { id: ldapId, name: username },
+            })
+
+            if (this._groupsConfig?.synchronize) {
+              await this._synchronizeGroups(user)
+            }
+          }
+
           return { username }
```

Comment:
The user has already been authenticated by the plugin here, we only need to return the user id.